seametrics/fo_utils/utils.py

```diff
@@ -192,12 +225,136 @@ def get_field_name(view: fo.DatasetView,
         list: List of values.
     """
 
-    if view.media_type == 'video':
+    if view.media_type == "video":
         if is_gt:
             return f"frames.{field_name}"
         else:
             return f"frames[].{field_name}"
-    elif view.media_type == 'image':
+    elif view.media_type == "image":
         return field_name
     else:
-        raise ValueError(f"Unsupported media type: {view.media_type}")
\ No newline at end of file
+        raise ValueError(f"Unsupported media type: {view.media_type}")
+
+
+def _add_batch(metric, data, model=None):
+    predictions, references = [], []
+
+    if model is None:
+        model = data["models"][0]
+
+    for sequence in data["sequence_list"]:
+        seq_data = data["sequences"][sequence]
+        gt_normalized = seq_data[data["gt_field_name"]]  # shape: (n_frames, m_gts)
+        pred_normalized = seq_data[model]  # shape: (n_frames, l_preds)
+        img_res = seq_data["resolution"]  # (h, w)
+        for gt_frame, pred_frame in zip(
+            gt_normalized, pred_normalized
+        ):  # iterate over all frame
+            processed_pred = _fo_dets_to_metrics_dict(
+                fo_dets=pred_frame, w=img_res[1], h=img_res[0], include_scores=True
+            )
+            processed_gt = _fo_dets_to_metrics_dict(
+                fo_dets=gt_frame,
+                w=img_res[1],
+                h=img_res[0],
+                include_scores=False,
+                include_areas=True,
+                gt_data=True,
+            )
+            predictions.append(processed_pred[0]["boxes"].tolist())
+            references.append(processed_gt[0]["boxes"].tolist())
+
+            # where the magic happens: update metric with data from current frame
+
+            metric.coco_metric.update(processed_pred, processed_gt)
+
+    return metric, predictions, references
+
+
+def _fo_dets_to_metrics_dict(
+    fo_dets: list,
+    w: int,
+    h: int,
+    include_scores: bool = False,
+    include_areas: bool = False,
+    gt_data: bool = False,
```

Comment:
```suggestion
    is_ground_truth: bool,
```
f the bboxes are ground truth, we should return the area and not the scores, and viceversa. I think we can get rid of the `include_scores` and `include_area` to simplify the code.

Reply:
I have kept only the is_ground_truth flage